clients/drcachesim/tracer/output.cpp

```diff
@@ -1340,11 +1347,19 @@ exit_thread_io(void *drcontext)
             BUF_PTR(data->seg_base), dr_get_thread_id(drcontext));
 
         ptr_int_t window = get_local_window(data);
-        process_and_output_buffer(drcontext,
-                                  /* If this thread already wrote some data, include
-                                   * its exit even if we're over a size limit.
-                                   */
-                                  data->bytes_written > 0);
+        if (has_tracing_windows()) {
+            if (is_in_tracing_mode(tracing_mode.load(std::memory_order_acquire)) &&
+                get_local_window(data) == window) {
+                process_and_output_buffer(
+                    drcontext,
+                    /* If this thread already wrote some data, include
+                     * its exit even if we're over a size limit.
+                     */
+                    data->bytes_written > 0);
+            }
+        } else {
+            process_and_output_buffer(drcontext, data->bytes_written > 0);
```

Comment:
Better to have just one call and a merged condition, to share the call and its 1355-1357 comment.

Reply:
Merged the condition. Made it clear that the extra check to `is_in_tracing_mode()` is used when `is_in_tracing_mode()`